### PR TITLE
Dockerfile beautification + upgrade

### PR DESCRIPTION
**Pull request by lluiscampos**
* [Dockerfile.daemon] Minor beautifications and optimizations
* [Dockerfile.daemon] Upgrade base image to ubuntu:20.04

### CODE REVIEW

**Comment by lluiscampos** [comment]
@mzedel Except for a tiny modification, I committed the file as you pasted it to me in Slack. Thanks!

File: tests/Dockerfile.daemon [hunk 1]
```diff
@@ -5,29 +5,30 @@ FROM ubuntu:20.04 AS build
...
-RUN sh -c 'echo Port 8822 >> /etc/ssh/sshd_config'
-
-COPY --from=build /mender-install/ /
-
-RUN sh -c 'mkdir -p /var/lib/mender && echo device_type=docker-client > /var/lib/mender/device_type'
-RUN sh -c 'echo artifact_name=original > /etc/mender/artifact_info'
+RUN echo 'PermitEmptyPasswords yes\n\
+PermitRootLogin yes\n\
+Port 22\n\
+Port 8822\n' >> /etc/ssh/sshd_config
```
Comment (kacf) [inline tests/Dockerfile.daemon:26]: Are you sure that this works? `echo` does not translate `\n` sequences, so you end up with a literal backslash and 'n'. That is, unless Docker translates it.

**Review by kacf — COMMENTED** [review]
Good cleanup!

File: tests/Dockerfile.daemon (hunk 1 above)
Comment (lluiscampos) [reply tests/Dockerfile.daemon:26]: This part actually works - I just verified that the lines as correctly appended to `sshd_config`. But something else is broken: the three integration tests that use this image are all failing :)

Investigating...

**Comment by lluiscampos** [comment]
>[...] But something else is broken: the three integration tests that use this image are all failing :)

Some paths were missing, like `/usr/share/mender`. Fixed now :+1:

File: tests/Dockerfile.daemon [hunk 2]
```diff
@@ -18,16 +20,18 @@ RUN apt update && apt install -y openssh-server
...
+
+COPY --from=build /mender-install/usr/bin/mender /usr/bin/mender
+COPY --from=build /mender-install/etc/mender /etc/mender
+COPY --from=build /mender-install/usr/share/mender /usr/share/mender
+COPY --from=build /mender-install/usr/share/doc/mender-client /usr/share/doc/mender-client
+COPY --from=build /mender-install/lib/systemd/system/mender-client.service /lib/systemd/system/mender-client.service
+COPY --from=build /mender-install/usr/share/dbus-1/system.d/io.mender.AuthenticationManager.conf /usr/share/dbus-1/system.d/io.mender.AuthenticationManager.conf
```
Comment (kacf) [inline tests/Dockerfile.daemon:33]: I guess you have a reason, but this is very cumbersome to maintain. Why doesn't the original `COPY --from=build /mender-install/ /` cut it?

**Review by kacf — APPROVED** [review]

File: tests/Dockerfile.daemon (hunk 2 above)
Comment (mzedel) [reply tests/Dockerfile.daemon:33]: that will just copy the changed files from the build stage - so everything else (e.g. `ls`, `sh`...) is gone and I have never seen the same behaviour for another image...
from what I found on looking into the ubuntu dockerfile, I _think_ the reason is a change in the ubuntu image build process (they at least planned to change to `buildah`), but couldn't track down the build steps and thus settled for the `COPY` approach...

Comment (kacf) [reply tests/Dockerfile.daemon:33]: Ah, I see. It's less like `cp` and more like `rsync -r --delete`. :-)

Comment (mzedel) [reply tests/Dockerfile.daemon:33]: unfortunately yes... but it should be more like `cp` (and it works that way in other repos + was with the `18.04` image) 🤷‍♂️

**Review by mzedel — APPROVED** [review]

File: tests/Dockerfile.daemon (hunk 2 above)
Comment (lluiscampos) [reply tests/Dockerfile.daemon:33]: @kacf Exactly. IOW, the real motivation to do this was to be able to upgrade to 20.04 image... I hope this does not bit us back in the near future when adding new directories or files to the Mender installation.

**Pull request merged**